src/ConfigManager.h

```diff
@@ -175,6 +175,9 @@ class ConfigManager {
   void startWebserver();
   void stopWebserver();
   void save();
+  void startAP();
```

Comment:
Is making this public a good idea? Perhaps I don't fully understand the use case. Also remember that we do not clean up between these functions, we expect a restart when switching from AP to API. By exposing them we allow you to do `startAP` followed by `startApi` which might do bad things, I never tested.

Reply:
All valid points, maybe we validate that there is not an existing wifi connection and change this to `bool starrtAP()` and return false when it bombs, true when it establishes?

I currently check that it is not in station mode ... and then I want to be able to use the configs to set the ap name/password .. currently I dont think I can do this. 

```
    wifiConnected = configManager.getMode() == station;
    //enable brownout detector

    if (!wifiConnected) {
      WiFi.softAPdisconnect();
      configManager.setAPName(config.deviceName);
      configManager.setAPPassword(config.apPassword);
      configManager.startAP();
      Serial.println(config.deviceName);
    }
```

Would making [readConfig](https://github.com/nrwiersma/ConfigManager/blob/master/src/ConfigManager.h#L247) public be a better option? The caller could then try to get configs before the wifi stuff happens, and then allow it to be set from there?

Reply:
exposing reading comes with its own issues, as you need to establish some things first

```
 void begin(T& config) {
    this->config = &config;
    this->configSize = sizeof(T);

    EEPROM.begin(CONFIG_OFFSET + this->configSize);

    setup();
  }
```

Reply:
The other option is to have some defined config names for these type of things `apName` `apPassword` and check for those. The question is, what wins those or `setAPName()` ?

Reply:
@nrwiersma I can also now dropped this. Turns out trying to use the device in AP mode and push a bunch of data seems to not make for a happy ESP chip .. Can kill this, and leave notes for future me to work around when/if I ever want to re-expose the AP or use name/pass from configs ...